### PR TITLE
fixes #2308 - add option to select cpumode in libvirt

### PR DESCRIPTION
**Pull request by cbergmann**
Hi,

this is my first try in exposing cpu mode in libvirt fog. 
I would be glad if this could be include in mainline oder you would point out some mistakes that I made.

I know of one bug that I Don't know how to fix:
If you select something other than 'default' as the default cpu mode in the settings. Then set the mode to 'default' while creating the Host. If you then edit the host the default from the settings will be shown and submitted.

### CODE REVIEW

**Comment by cbergmann** [comment]
> If you select something other than 'default' as the default cpu mode in the settings. Then set the mode to 'default' while creating the Host. If you then edit the host the default from the settings will be shown and submitted.

One Solution would be to remove the setting altogether and "force default " to 'default'. A default value can be set on the hostgroup level. Should I remove the setting?

**Comment by xprazak2** [comment]
The main problem here is that you fall back on the value from setting when the attribute is `nil` and when you choose 'default', you are assigning `nil` in `cpu_mode=(cpumode)` method.

You can try to remove the setting, however I think you will hit the same problem with hostgroups: If you set anything other than 'default' for hostgroup, you will not be able to override it in host to 'default'. This is a behaviour we currently have for other attributes as well.

But removing the setting is probably a good idea anyway, as it is tied to a specific provider while the provisioning settings are quite general.

File: app/models/setting/provisioning.rb [hunk 1]
```diff
@@ -74,7 +82,8 @@ def self.default_settings
         N_("Exclude pattern for all types of imported facts (rhsm, puppet e.t.c.). Those facts won't be stored in foreman's database. You can use * wildcard to match names with indexes e.g. macvtap*"),
         default_excluded_facts,
         N_('Exclude pattern for facts stored in foreman')
-      )
+      ),
+      self.set('libvirt_default_cpu_mode', N_("Default cpu mode for libvirt."), 'custom', N_("Default Libvirt CPU mode."), nil, {:collection => Proc.new {CPU_MODES} })
```
Comment (xprazak2) [inline app/models/setting/provisioning.rb]: 'custom' is commented out on l.38, so maybe this should be 'default'?

**Review by xprazak2 — APPROVED** [review]
Works nicely

**Comment by xprazak2** [comment]
Thanks!

**Comment by lzap** [comment]
Nice work thanks!

**Pull request merged**